### ISSUE TITLE
MeshTexture behaves poorly when used as icon on Button

### ISSUE THREAD

**Issue by jgillich** [godotengine/godot#63801]
### Godot version

5ecd61a3150f87a433dd5a7ff49a842b72516640

### System information

Fedora Silverblue 36, Ryzen 2400G

### Issue description

There are several issues when using `MeshTexture` as icon in a `Button`.

* After assigning texture as icon, editor preview does not update until another unrelated property on button is changed
* Icon is completely misaligned
* `expand_icon` setting hides texture (or makes it zero size)

Demonstration:

https://user-images.githubusercontent.com/347965/182269987-3dcb56bc-2640-4202-a936-2772dc0547b5.mov

### Steps to reproduce

1. Add button
2. Assign a MeshTexture as icon

**Comment by Rindbee** [comment]
This is because the `changed` signal of the texture is not connected in `Button`.

**Comment by AThousandShips** [comment]
Will work on a fix for this

**Comment by AThousandShips** [comment]
The `expand_icon` part of this is covered by #59202

**Comment by DenisMVP** [comment]
Found the same issue with TextureRect. Also it only updates when changing the TextureRect Visibility
![image](https://github.com/godotengine/godot/assets/120058545/2b16f6f3-0dd1-4d5f-85aa-e8381c127b80)

**Comment by AThousandShips** [comment]
What version are you on?

**Comment by DenisMVP** [comment]
I found a similar issue with Godot 4.1 using TextureRect. I opened an issue https://github.com/godotengine/godot/issues/81213

**Comment by P3NG00** [comment]
I just ran into this issue. I wanted a way to add a color to a texture button. I couldn't find a way to add color to a Texture2D so I tried using MeshTexture with a QuadMesh so I could use the material and color of the texture mesh. The same behavior described in Denis' post still occurs in Godot 4.3. As well as the TextureMesh ignores the QuadMesh's set material. It will only take the texture placed in the MeshTexture.
